nwb_conversion_tools/tools/spikeinterface/spikeinterface.py

```diff
@@ -253,36 +253,23 @@ def add_electrodes(
         object to ignore when writing to the NWBFile.
     """
     assert isinstance(nwbfile, pynwb.NWBFile), "'nwbfile' should be of type pynwb.NWBFile"
+
     if isinstance(recording, RecordingExtractor):
-        checked_recording = OldToNewRecording(oldapi_recording_extractor=recording)
-    else:
-        checked_recording = recording
+        recording = OldToNewRecording(oldapi_recording_extractor=recording)
```

Comment:
Just so we're 100% on this behavior, what will happen now is that any time anyone uses a `recording` that is a `RecordingExtractor` (old type) in conjunction with `add_electrode` or `write_recording`, the `recording` they passed into the function will be modified in-place to become a `BaseRecording` mapped via the `OldToNew` class.

Personally not a fan of behavior like that because it can produce very unexpected results when/if the user isn't expecting it; _e.g._, if they had any lines after calling `add_electrodes` or `write_recording` in their script that assumed the recording to still be a `RecordingExtractor`, those lines will now fail due to the API differences.

I'd suggest restoring the `checked_recording`, but also feel free to make it a general function used across all the methods here to reduce amount of code.

Reply:
Aha!
This was my omission of understanding. Python passes most stuff by reference so this means that the outer-scope recording object will be modified. You are 100 % correct Cody.